src/library/clementine/clementinedbconnection.cpp

```diff
@@ -0,0 +1,182 @@
+        : m_database(),
+          m_pTrackCollectionManager(pTrackCollection) {
+}
+
+ClementineDbConnection::~ClementineDbConnection() {
+    qDebug() << "Close Clementine database";
+    m_database.close();
+}
+
+bool ClementineDbConnection::open(const QString& databaseFile) {
+    m_database = QSqlDatabase::addDatabase("QSQLITE", "CLEMENTINE_DB_CONNECTION");
+    m_database.setHostName("localhost");
+    m_database.setDatabaseName(databaseFile);
+    m_database.setConnectOptions("SQLITE_OPEN_READONLY");
+
+    //Open the database connection in this thread.
+    if (!m_database.open()) {
+        m_database.setConnectOptions(); // clear options
+        qWarning() << "Failed to open Clementine database." << m_database.lastError();
+        return false;
+    } else {
+        qDebug() << "Successful opened Clementine database";
+        return true;
+    }
+}
+
+QList<ClementinePlaylist> ClementineDbConnection::getPlaylists() const {
+    QList<ClementinePlaylist> list;
+    ClementinePlaylist playlist;
+
+    QSqlQuery query(m_database);
+    query.prepare("SELECT ROWID, playlists.name FROM playlists ORDER BY playlists.name");
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return {};
+    }
+    while (query.next()) {
+        playlist.playlistId = query.value(0).toString();
+        playlist.name = query.value(1).toString();
+        list.append(playlist);
+    }
+    return list;
+}
+
+QList<ClementinePlaylistEntry>
+ClementineDbConnection::getPlaylistEntries(int playlistId) const {
+    PerformanceTimer time;
+    time.start();
+
+    QList<ClementinePlaylistEntry> list;
+    ClementinePlaylistEntry entry;
+
+    QSqlQuery query(m_database);
+    query.setForwardOnly(true); // Saves about 50% time
+    query.prepare(
+            "SELECT "
+            "ROWID, "
+            "playlist_items.title, "      // 1
+            "playlist_items.filename, "   // 2
+            "playlist_items.length, "     // 3
+            "playlist_items.artist, "     // 4
+            "playlist_items.year, "       // 5
+            "playlist_items.album, "      // 6
+            "playlist_items.rating, "     // 7
+            "playlist_items.genre, "      // 8
+            "playlist_items.track, "      // 9
+            "playlist_items.bpm, "        // 10
+            "playlist_items.bitrate, "    // 11
+            "playlist_items.comment, "    // 12
+            "playlist_items.playcount, "  // 13
+            "playlist_items.composer, "   // 14
+            "playlist_items.grouping, "   // 15
+            "playlist_items.type, "       // 16
+            "playlist_items.albumartist " // 17
+            "FROM playlist_items "
+            "WHERE playlist_items.playlist = :playlistId");
+    query.bindValue(":playlistId", playlistId);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return {};
+    }
+    while (query.next()) {
+        QString type = query.value(16).toString();
+        if (type != "File")
```

Comment:
Found the issue. This needs to be "Library", not "File" for me (Clementine 1.4 rc2).

Also, please *always* use braces. And I think this can be a QLatin1String.

Reply:
I have also Clementine 1.4 rc2. I tried to recreate your issue by deleting my old library and let clementine create a new one.

The query as written in the code gets the tracks for a specific playlist. The output looks as following for a new playlist with 5 tracks:
 ```
sqlite> SELECT * FROM schema_version ;
51
sqlite> SELECT ROWID, playlist_items.type FROM playlist_items WHERE playlist_items.playlist = 1;
1|File
2|File
3|File
4|File
5|File
  ```
I cant reproduce the problem for now with my clementine version.

Reply:
A i think i know the issue now: i used clementine more like a playlist player, instead of using the clementine library feature. Because if i add all the songs to the library, the type changes from "File" to "Library".

Will work on a fix now...

Reply:
Ok, cool. The library feature should support both.

Reply:
The other thing you mentioned. I am not deeply into string encoding, so its like that clementine would save every string in the DB with different encoding on a other machine with a different encoding?

Would
`
if (type == QString("File"))
` 
fix this already?

Reply:
> The other thing you mentioned. I am not deeply into string encoding, so its like that clementine would save every string in the DB with different encoding on a other machine with a different encoding?

I don't think this is an issue. Apparently the clementine DB uses UTF-8:
```
$ sqlite3 ~/.config/Clementine/clementine.db
SQLite version 3.34.0 2020-12-01 16:14:00
Enter ".help" for usage hints.
sqlite> PRAGMA encoding;
UTF-8
```

> Would `if (type == QString("File"))` fix this already?

Use `QLatin1String` instead of `QString` for such a comparison for performance reasons.

Reply:
`QLatin1String` didnt work out for me , hope `QString` is also okay?